### DISCUSSION TITLE
Deleted error

### DISCUSSION THREAD

**Discussion by tailorck** [Budibase/budibase#6518] [Help] [answered]
I used to get this error at the top of every page on my app that said `Internal Server Error`. Browsing prior discussions, I found that it may be because of a broken database connection. So I deleted my database connection, reconnected it, and added back all my queries. Now I get this error that says `deleted`. Can anyone help me understand what is going on?
<img width="1728" alt="image" src="https://user-images.githubusercontent.com/7613002/176816319-3a02736c-c11e-46aa-9b9a-7265071eab88.png">

**Comment by melohagan** [comment]
Hi @tailorck 

The "deleted" message means that something in your design has been deleted from CouchDB. 

Could you open the network tab and share any detailed error responses please. (Fn+F12 on Chrome)

**Reply by NicolasFabre81** [reply]
I have the same problem after having changed the database connection.
Even If I re-validate my data providers, I still get the "deleted" message on some screens.

I do not see anything in the chrome dev tools but maybe I do not know where to look at :
<img width="714" alt="image" src="https://user-images.githubusercontent.com/28592069/186697670-09a44737-77b2-4fa8-9190-c8d761ab2d5e.png">

CC @mielune

**Comment by NicolasFabre81** [comment]
Here I think it shows the 2 deleted data sources causing the 2 error messages.
But how to hard delete these so that BB does not use them anymore ?
In the design screen everything shows OK
<img width="717" alt="image" src="https://user-images.githubusercontent.com/28592069/186700793-3aae3eed-3071-4470-8fd5-aadcd73d6a1b.png">

**Reply by NicolasFabre81** [reply]
I think a solution could be to export the app, make some corrections in the txt file (delete the old datasources ID....) and the re-import the app.

But the txt app format is quite undrinkable ...

**Comment by NicolasFabre81** [comment, marked as answer]
I confirm that was the solution !
1/ Check with the chrome dev tools the ID of the problematic datasources
2/ Note the good ID to replace it with
3/ Export the App in txt format
4/ In Notepad, replace the 1/ by the 2/, for example datasource_plus_acc14676a87a4c3ba0fb83fa9c9a7a50__echanges by datasource_plus_d0c9f142cc364f22819b80c27735a959__echanges
5/ Re-import your app : no more deleted messages !

The problem seems to affect DB actions related to buttons (SAVE ROW ...) whereas data providers seem to be correctly managed when changing the data source

**Reply by melohagan** [reply]
Brilliant, thanks very much for sharing Nicolas!
